realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -1056,4 +1056,27 @@ public void testWriteEncryptedCopy() throws Exception {
         }
     }
 
+    public void testWrongKeyShouldFail() {
+        final String WRONG_KEY_REALM = "wrong-key-realm.realm";
+
+        // Wrong key size
+        try {
+            Realm.getInstance(getContext(), WRONG_KEY_REALM, new byte[63]);
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } catch (Exception ignored) {
```

Comment:
I'm trying to make sure we throw the right exception and be explicit about it. If I don't catch exception and then somebody adds a "thows Exception" to the declaration of the testcase (as it is normally, even autogenerated test templates by Android Studio) the test would pass when it should not.